### PR TITLE
Symphosize improvements

### PR DESCRIPTION
**Pull request by Symphosize**
This includes our changes to the Queue plugin.  There are two main changes included with this pull request.
1. We added the concept of priority to queued tasks.  By default all tasks will have a priority of 5.  But you can specify a lower number like 3 and this will place the task in front of those with 5.

`$this->QueuedTask->nextPriority(2)->createJob('YourSuperCoolTask', [
'some_id' => '12345',
]);`
1. We have had cases where a task that is working with a remote api fails because of exceptions raised when the API is down.  We should be catching these exceptions in the task and handle failure there.  But this is a fail safe.  We added a try catch block that will log any exception making it up to the runworker method. then mark the task as failed and move on with out causing the worker to die.

I am open to suggestion on how we can make this a useful contribution to the plugin.  Any feedback is welcome.

### CODE REVIEW

**Comment by Symphosize** [comment]
I see the tests are failing because of our missing `priority` column,  We have included a cakeDC migration, but I don't think these were used by this repo.  Should I also include a schema snapshot?

**Comment by dereuromark** [comment]
I like the changes.
The tests use fixtures. Those need updating.

**Comment by Symphosize** [comment]
I'll update the fixtures tomorrow, I'll issue a slight change to our exception catch to use `\Exception $e` instead of just `Exception $e` I just learned that when code is name spaced plain old Exception won't catch :-(  I will also take a look at the tests and see what I can do about testing the `nextPriority` method

**Comment by dereuromark** [comment]
But this 2.x code is not namespaced, thus it should not matter.

**Comment by Symphosize** [comment]
However the libs we use are often namespaced.  For example I am having trouble with the Google API throwing and Error, and using `catch( Exception $e ) {} ` fails to catch the error... it has to be `catch( \Exception $e ) {} `  What I don't know is if I need both catch blocks or if the namespaced version will work as a catch all for any exception bubbling up to the worker.  Just trying to put up a firewall so that our workers don't fall over in the middle of the night.

File: Config/Migration/1453822680_adding_priority_field_to_queue_tasks.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		return true;
+	}
+
+/**
+ * After migration callback
+ *
+ * @param string $direction Direction of migration process (up or down)
+ * @return bool Should process continue
+ */
+	public function after($direction) {
```
Comment (dereuromark) [inline Config/Migration/1453822680_adding_priority_field_to_queue_tasks.php]: Would be great if you could run the sniffer over this to auto-fix the comment indentation.

File: Model/QueuedTask.php [hunk 2]
```diff
@@ -11,6 +11,8 @@
  */
 class QueuedTask extends QueueAppModel {
 
+	public $_next_priority = 5;
```
Comment (dereuromark) [inline Model/QueuedTask.php]: $nextPriority

**Comment by Symphosize** [comment]
Just a note, that I am still planning on making these changes.  Will need a couple more days

**Comment by dereuromark** [comment]
Cool

File: Model/QueuedTask.php [hunk 3]
```diff
@@ -76,8 +84,10 @@ public function createJob($jobName, $data = null, $notBefore = null, $group = nu
 			'jobtype' => $jobName,
 			'data' => serialize($data),
 			'group' => $group,
-			'reference' => $reference
+			'reference' => $reference,
+			'priority' => $this->_next_priority
 		];
+		$this->_next_priority = 5; // reset to default;
```
Comment (dereuromark) [inline Model/QueuedTask.php]: _nextPriority

File: Config/Migration/1453822680_adding_priority_field_to_queue_tasks.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
+<?php
+class AddingPriorityFieldToQueueTasks extends CakeMigration {
+
+/**
+ * Migration description
+ *
+ * @var string
+ */
```
Comment (dereuromark) [inline Config/Migration/1453822680_adding_priority_field_to_queue_tasks.php]: Indentation is off.

**Pull request merged**